Backend - Upgraded MLMD client to fix Metadata Writer

Ark-kun opened the pull request:
There was a backwards-incompatible MLMD server-side change that caused the get_artifacts_by_uri API to start silently returning empty lists of artifacts.
Fixes https://github.com/kubeflow/pipelines/issues/3656

numerology commented:
Perhaps we shall add upgrading this dependency as a check item of releasing process. WDYT? @Ark-kun

Ark-kun commented:
> Perhaps we shall add upgrading this dependency as a check item of releasing process. WDYT? @Ark-kun

Added a section in the Release Playbook and also added instructions in all relevant files: https://github.com/kubeflow/pipelines/pull/3658

Bobgy commented:
Thanks!
/lgtm

Ark-kun commented:
/approve

Ark-kun commented:
/hold
Travis...

Ark-kun commented:
/hold cancel
